scipy/_lib/doccer.py

```diff
@@ -202,73 +265,106 @@ def indentcount_lines(lines):
     return indentno
 
 
-def filldoc(docdict, unindent_params=True):
-    ''' Return docstring decorator using docdict variable dictionary
+def filldoc(docdict: dict[str, str], unindent_params: bool = True) -> Decorator[R]:
+    """Return docstring decorator using docdict variable dictionary.
 
     Parameters
     ----------
-    docdict : dictionary
-        dictionary containing name, docstring fragment pairs
-    unindent_params : {False, True}, boolean, optional
-        If True, strip common indentation from all parameters in
-        docdict
+    docdict : dict[str, str]
+        A dictionary containing name, docstring fragment pairs.
+    unindent_params : bool, optional
+        If True, strip common indentation from all parameters in docdict.
+        Default is False.
 
     Returns
     -------
-    decfunc : function
-        decorator that applies dictionary to input function docstring
-
-    '''
+    Decorator[R]
```

Comment:
`decfunc : function` is correct; the numpydoc format is `name : type`. Can you undo changes like these?